User (Email, Nickname) validate 뷰에서 request body 받는 부분, response 수정

cafitac opened the pull request:
member/views.py
 - MOD: request.POST.get('email') -> request.data['email']
 - MOD: request.POST.get('nickname') -> request.data['nickname']
 - ADD: serializer.is_valid() 작업 후 message에 따라 APIException 작동
 - MOD: (Email, Nickname) 닉네임 사용 여부의 response status code가 반대로 되어있어 변경

member/exceptions.py
 - ADD: email_validation에 대한 APIException 작성 (Invalid, Required, Unique, Useful)
 - ADD: nickname_validation에 대한 APIException 작성 (Required, MaxLength, Unique, Useful)

PS: 저도 현재 방법이 마음에 들지 않은데 리뷰를 받고 고치는게 나아서 PR 넣어봤습니다...ㅠㅠ

hanseongd commented on scheduler/member/views.py:
```diff
@@ -89,32 +89,32 @@ def search(self, request):
 @api_view(['POST'])
 def user_email_validate(request):
-    email = request.POST.get('email')
+    email = request.data['email']
```
KEY ERROR 처리 필요

cafitac replied:
앗 넵!!

hanseongd reviewed (commented):
email필드랑 nickname 필드가 unique 처리 되어있다면 두 개의 함수 전부 serializer로 처리 가능해

Commit:
User (Email, Nickname) validate 뷰에서 request body 받는 부분, response 수정

member/views.py
 - MOD: request.POST.get('email') -> request.data['email']
 - MOD: request.POST.get('nickname') -> request.data['nickname']
 - ADD: serializer.is_valid() 작업 후 message에 따라 APIException 작동
 - MOD: (Email, Nickname) 닉네임 사용 여부의 response status code가 반대로 되어있어 변경

member/exceptions.py
 - ADD: email_validation에 대한 APIException 작성 (Invalid, Required, Unique, Useful)
 - ADD: nickname_validation에 대한 APIException 작성 (Required, MaxLength, Unique, Useful)

Commit:
User (Email, Nickname) validate http method 변경

member/views.py
 - MOD: user_email_validate http method (POST -> GET)으로 변경
 - MOD: user_nickname_validate http method (POST -> GET)으로 변경
 - MOD: (user_email_validate, user_nickname_validate) 데이터 querystring으로 받는 것으로 변경

member/exceptions.py
 - MOD: EmailRequired -> EmailBlank 로 이름 변경
 - MOD: NicknameRequired -> NicknameBlank 로 이름 변경

hanseongd commented on scheduler/member/views.py:
```diff
@@ -111,16 +113,20 @@ def user_email_validate(request):
     try:
         serializer.is_valid(raise_exception=True)
     except ValidationError:
+        print(serializer.errors)
```
print 문 제거

hanseongd commented on scheduler/member/exceptions.py:
```diff
@@ -8,10 +8,10 @@ class EmailInvalid(APIException):
 
-class EmailRequired(APIException):
+class EmailBlank(APIException):
     status_code = status.HTTP_202_ACCEPTED
```
에러 내용인데 status code 202로 하기로 한거야?

cafitac replied:
음.. 코드를 뭐로 할까요?? 왜 202로 했는지는 잘 기억이 안 나는데 4xx로 바꾸는게 좋아보여요

hanseongd replied:
404 code

Commit:
Exception status code 변경, print문 삭제

member/views.py
 - DEL: serializer error print문 제거

member/exceptions.py
 - MOD: EmailBlank status 202 -> 404 로 변경
 - MOD: NicknameBlank status 202 -> 404로 변경

Commit:
내 친구 리스트 뷰 추가

member/views.py
 - ADD: FriendRequestViewSet에 친구 리스트를 가져오는 friend_list action 추가

hanseongd commented on scheduler/member/views.py:
```diff
@@ -189,4 +189,28 @@ def create(self, request, *args, **kwargs):
+                }
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        friends_serializer = []
```
friends_serializer = user.friends.values_list('id', flat=True)
이런식으로 바꾸면 될거같은데
values_list랑 values에 대해서 찾아보면 금방 이해 될거야

cafitac replied:
넵넵 이해됐어욥

Commit:
User 친구 리스트 불러오는 로직 수정

member/views.py
 - MOD: friend_list action 뷰를 FriendRequestViewSet에서 UserProfileViewSet으로 이동
 - MOD: values_list를 활용하여 직접 for문으로 친구 pk 값을 저장하는 로직 개선
